google-cloud-logging/src/main/java/com/google/cloud/logging/GaeFlexLoggingEnhancer.java

```diff
@@ -65,13 +70,29 @@ public GaeFlexLoggingEnhancer() {
 
   @Override
   public void enhanceMonitoredResource(Builder builder) {
-    if (gaeInstanceId != null) {
-      if (System.getenv("GAE_SERVICE") != null) {
-        builder.addLabel("module_id", System.getenv("GAE_SERVICE"));
-      }
-      if (System.getenv("GAE_VERSION") != null) {
-        builder.addLabel("version_id", System.getenv("GAE_VERSION"));
+    if (gaeInstanceId == null) {
+      return;
+    }
+    if (System.getenv("GAE_SERVICE") != null) {
+      builder.addLabel("module_id", System.getenv("GAE_SERVICE"));
+    }
+    if (System.getenv("GAE_VERSION") != null) {
+      builder.addLabel("version_id", System.getenv("GAE_VERSION"));
+    }
+    try {
+      HttpURLConnection metadata = (HttpURLConnection) new URL(
+          "http://metadata.google.internal/computeMetadata/v1/instance/zone").openConnection();
+      metadata.setRequestProperty("Metadata-Flavor", "Google");
```

Comment:
I followed the documentation [here](https://cloud.google.com/compute/docs/storing-retrieving-metadata), which says

> Note: Previously, the `X-Google-Metadata-Request: True` header was required in requests. Both of these headers are still supported but it is recommended that you use the `Metadata-Flavor` header rather than the `X-Google-Metadata-Request: True`header.

I have verified that it works with and does not work without the `Metadata-Flavor` header.

Should we correct the other usage in this PR?